app/Filament/Resources/UserResource.php

```diff
@@ -4,47 +4,132 @@
 use Filament\Forms\Form;
 use Filament\Resources\Resource;
+use Filament\Tables\Actions\EditAction;
+use Filament\Tables\Columns\SpatieMediaLibraryImageColumn;
+use Filament\Tables\Columns\TextColumn;
 use Filament\Tables\Table;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Str;
 
 class UserResource extends Resource
 {
     protected static ?string $model = User::class;
 
     protected static ?string $navigationIcon = 'heroicon-o-user';
 
-    protected static bool $shouldRegisterNavigation = false;
-
     public static function form(Form $form): Form
     {
         return $form
             ->schema([
-                //
+                TextInput::make('email')->required()->columnSpan('full')->email(),
+                TextInput::make('password')->columnSpan('full')->password()
+                    ->dehydrateStateUsing(fn (string $state): string => Hash::make($state))
+                    ->dehydrated(fn (?string $state): bool => filled($state))
+                    ->required(fn (string $operation): bool => $operation === 'create')
+                    ->autocomplete('new-password'),
+                TextInput::make('display_name')->columnSpan('full'),
+                Select::make('role')
+                    ->options(function () {
+                        $options = [
+                            Role::Editor->value => Str::title(Role::Editor->value),
+                        ];
+
+                        /** @var User */
+                        $user = auth()->user();
+
+                        if ($user->hasRole(Role::Admin->value)) {
+                            $options[Role::Admin->value] = Str::title(Role::Admin->value);
+                        } elseif ($user->hasRole(Role::Superadmin->value)) {
+                            $options[Role::Admin->value] = Str::title(Role::Admin->value);
+                            $options[Role::Superadmin->value] = Str::title(Role::Superadmin->value);
+                        }
+
+                        return $options;
+                    })
+                    ->afterStateHydrated(function (Select $component, $state) use ($form) {
+                        /** @var User|null */
+                        $user = $form->getRecord();
+                        /** @var RoleModel|null $role */
+                        $role = $user?->roles()->first();
+
+                        if ($role !== null) {
+                            $component->state($role->name);
+                        }
+                    })
+                    ->default(Role::Editor->value)
+                    ->required(),
+
+                SpatieMediaLibraryFileUpload::make('avatar')
+                    ->collection('avatar')
+                    ->columnSpan('full')
+                    ->image()
+                    ->imageEditor()
+                    ->imageCropAspectRatio('1:1'),
+
+                CheckboxList::make('permissions')
```

Comment:
can you add a subtitle/helper text to indicate that these permissions will be on an individual basis and in addition to the ones from the assigned role, e.g:
"Selected permissions here will be applied in addition to the assigned user role"